src/scripting/lua_map.cc

```diff
@@ -2178,96 +2201,116 @@ int LuaMapObjectDescription::get_name(lua_State* L) {
+            for all buildings
+          * :class:`constructionsite <ConstructionSiteDescription>`,
+            an actual building is being constructed here,
+          * :class:`dismantlesite <DismantleSiteDescription>`, an
+            actual building is being dismantled here,
+          * :class:`warehouse <WarehouseDescription>`, a warehouse
+            can store wares and workers. Headquarters and ports are
+            special types of warehouses, but they belong to the same
+            class,
+          * :class:`militarysite <MilitarySiteDescription>`, a
+            building manned by soldiers to expand a tribe's territory,
+          * :class:`productionsite <ProductionSiteDescription>`, the
+            most common type of building, which can produce wares,
+          * :class:`trainingsite <TrainingSiteDescription>`, a
+            specialized productionsite for improving soldiers.
+
+        * **Other Immovables:** Specialized immovables that aren't buildings:
+
+          * :class:`flag <FlagDescription>`, a flag that can hold
+            wares for transport,
+          * :class:`road <RoadDescription>`, a road or waterway
+            connecting two flags,
+          * :class:`portdock <PortdockDescription>`, a 'parking space'
+            on water terrain where ships can load/unload wares and
+            workers. A portdock is invisible to the player and one is
+            automatically placed next to each port building.
+
+      * **Abstract:** These types are abstract map objects that are used by the engine and are
+        not visible on the map.
+
+        * :class:`battle <BattleDescription>`, holds information
+          about two soldiers in a fight,
+        * :class:`fleet <FleetDescription>`, holds information for
+          managing ships.
+
+      Example to fetch some information from a tribe's description:
+
+      .. code-block:: lua
+
+         -- get tribe description
+         local tribe_descr = wl.Game():get_tribe_description("barbarians")
+
+         -- get building descriptions of this tribe
+         local buildings = tribe_descr.buildings
+
+         -- iterate over all building descriptions
+         for i, building in ipairs(buildings) do
+            print(building.type_name, building.name)
+
+            -- filter military sites
+            if building.type_name == "militarysite" do
+               print(building.max_number_of_soldiers)
+            end
+         end
 */
 int LuaMapObjectDescription::get_type_name(lua_State* L) {
 	lua_pushstring(L, to_string(get()->type()));
 	return 1;
 }
 
 /* RST
-   .. method:: helptext(tribename)
+   .. method:: helptexts(tribename)
+
+      (RO) Returns a :class:`table` of helptexts if it exists for the given tribe, an
+      empty table otherwise.
 
-      Returns the tribe-specific helptexts for this object.
+      Keys are ``lore``, ``lore_author``, ``purpose``, ``note``, ``performance``, all of them
```

Comment:
Strictly, these only are the keys *commonly found*. You can also define `foobar="helloworld"` in the helptext definition in the units.lua, and this  should be present in the resulting table here.

Reply:
Since this [hadn't worked before](https://github.com/widelands/widelands/pull/4822) (ff) i am in favor of removing this entire function.

Reply:
Added a sentence that the table may can contain other keys as well.